src/core/Akka/Util/Internal/AtomicState.cs

```diff
@@ -196,6 +199,8 @@ public void Enter()
     /// </summary>
     public interface IAtomicState
     {
+        Exception LastCaughtException { get; }
```

Comment:
Since the `CircuitBreaker._currentState` is of type `AtomicState` (not `IAtomicState`) and this is `internal` anyway, we could just make `LastCaughtException` a protected internal field of `AtomicState` instead. Just thinking out loud...

Reply:
LastCaughtException is accessed by the circuit breaker, so it needs to be public, but yeah, it can be moved to `AtomicState`. Actually, `IAtomicState` is an unnescessary abstraction, since the abstract `AtomicState` is enough.